src/components/Articles.jsx

```diff
@@ -1,34 +1,73 @@
 import { getRequest } from "../utils/api";
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
+import { useSearchParams, Link } from "react-router-dom";
 import ArticleCard from "./ArticleCard";
 import Loading from "./Loading";
+import ArticlesSideBar from "./ArticlesSideBar";
 const Articles = ({
   listArticles,
   setlistArticles,
   isLoading,
   setIsLoading,
 }) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [currentTopic, setCurrentTopic] = useState(searchParams.get("topic"));
+  const [currentTopicDescription, setCurrentTopicDescription] = useState("");
+  const [topics, setTopics] = useState([]);
+
+  const getRequestURL = currentTopic
+    ? `/api/articles?topic=${currentTopic}`
+    : "/api/articles";
+  console.log(currentTopic);
+
   useEffect(() => {
-    getRequest("/api/articles")
+    getRequest(getRequestURL)
       .then(({ articles }) => {
         setlistArticles(articles);
         setIsLoading(false);
       })
       .catch((err) => console.log(err));
   }, []);
 
+  useEffect(() => {
+    getRequest("/api/topics").then(({ topics }) => {
+      setTopics(topics);
+
+      const currentTopicApi = topics.filter((apiTopic) => {
+        return apiTopic.slug === currentTopic;
+      });
+      setCurrentTopicDescription(currentTopicApi[0].description);
+    });
```

Comment:
This is really good logic, but I would recommend renaming the topics parameter here to something similar to responseTopics, just so its clear it isnt the same topics as the state defined at the top of this component.